src/utils/hooks/useFetchPublication.ts

```diff
@@ -0,0 +1,63 @@
+import { useState, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import Axios from 'axios';
+
+import { setNotification } from '../../redux/actions/notificationActions';
+import { NotificationVariant } from '../../types/notification.types';
+import { PublicationStatus, emptyPublication, FormikPublication } from '../../types/publication.types';
+import { getPublication } from '../../api/publicationApi';
+import { useHistory } from 'react-router-dom';
+import { RootStore } from '../../redux/reducers/rootReducer';
+import deepmerge from 'deepmerge';
+
+const useFetchPublication = (
+  identifier: string | undefined,
+  editMode: boolean = false,
+  closeForm?: () => void
+): [FormikPublication, boolean, (values: FormikPublication) => void] => {
+  const dispatch = useDispatch();
+  const [publication, setPublication] = useState<FormikPublication>(emptyPublication);
+  const [isLoading, setIsLoading] = useState(!!identifier);
+  const history = useHistory();
+  const user = useSelector((store: RootStore) => store.user);
+
+  const handleSetPublication = (values: FormikPublication) => {
+    setPublication(values);
+  };
+
+  useEffect(() => {
+    const cancelSource = Axios.CancelToken.source();
+
+    const fetchPublication = async () => {
+      const publication = await getPublication(identifier!, cancelSource.token);
+      if (publication) {
+        setIsLoading(false);
+        if (publication.error) {
+          dispatch(setNotification(publication.error, NotificationVariant.Error));
+          closeForm?.();
```

Comment:
et problem med egne fetchHooks er når vi må ta inn sånne ting som (trolig) bare trengs en bestemt plass, men som kompliserer hooken for all bruk. :/

Reply:
sikkert en fordel om vi prøve å legge ting utenfor hooken hvor det er mulig med en useEffect feks

Reply:
ja, men synes du det er bedre å returnere `publication.error` i hooken, og i `PublicationForm.tsx` sjekke på `publication.error` og å kalle `closeForm` i en useEffect der?

Reply:
vet ikke tbh, var mer en generell betraktning... men når jeg jobber med filopplasting nå, ville jeg legge til filene som ligger i publikasjonen i uppy med `uppy.addFile()`, slik at uppy kan nekte brukeren å legge til samme fil igjen etterpå. Hadde vi brukt "vanlig" useEffect ville jeg lagt det inn på linje 42 her, men jeg vil i hvert fall ikke dra inn noe sånt inn i en generell fetchHook, så vi må være litt obs på det

Reply:
ja, skjønner. kan gjøre et forsøk på å dra ut `closeForm`, også kan vi se om det blir bedre?

Reply:
du kan jo prøve, å se hva som ser best ut. your call